CTS1.rst

```diff
@@ -0,0 +1,666 @@
+
+  Transfer ::= (id: TokenID) (amount: TokenAmount) (from: Address) (to: Receiver) (data: AdditionalData)
+
+  TransferParameter ::= (n: Byte) (transfers: Transferⁿ)
+
+.. note::
+
+  Be aware of the smart contract parameter size limit of 1024 bytes.
+  Since the byte size of a single transfer can vary in size, this will limit the number of transfers can be included in the same function call.
+  Currently, with the smallest possible transfers, the parameter can contain 21 transfers and with the biggest possible transfer will take the whole parameter.
+
+.. _CTS-functions-transfer-receive-hook-parameter:
+
+Receive hook parameter
+~~~~~~~~~~~~~~~~~~~~~~
+
+The parameter for the receive function hook contains information about the transfer, the name of the token contract and some additional data bytes.
+
+It is serialized as: a :ref:`CTS-TokenID` (``id``), a :ref:`CTS-TokenAmount` (``amount``), the token owner address :ref:`CTS-Address` (``from``), the name of the token contract :ref:`CTS-ContractName` (``contract``) and :ref:`CTS-AdditionalData` (``data``)::
+
+  ReceiveHookParameter ::= (id: TokenID) (amount: TokenAmount) (from: Address) (contract: ContractName) (data: AdditionalData)
+
+Requirements
+~~~~~~~~~~~~
+
+- The list of transfers MUST be executed in order.
+- The contract function MUST reject if any of the transfers fails to be executed.
+- A transfer MUST fail if:
+
+  - The token balance of the ``from`` address is insufficient to do the transfer with error :ref:`INSUFFICIENT_FUNDS<CTS-rejection-errors>`.
+  - TokenID is unknown with error: :ref:`INVALID_TOKEN_ID<CTS-rejection-errors>`.
+
+- A transfer MUST non-strictly decrease the balance of the ``from`` address and non-strictly increase the balance of the ``to`` address or fail.
+- A transfer with the same address as ``from`` and ``to`` MUST be executed as a normal transfer.
+- A transfer of a token amount zero MUST be executed as a normal transfer.
+- A transfer of some amount of a token type MUST only transfer the exact amount of the given token type between balances.
+- A transfer of any amount of a token type to a contract address MUST call receive hook function on the receiving smart contract with a receive hook parameter :ref:`described above<CTS-functions-transfer-receive-hook-parameter>`
+- The contract function MUST reject if a receive hook function called on the contract receiving tokens rejects.
+
+.. _CTS-functions-updateOperator:
+
+``updateOperator``
+^^^^^^^^^^^^^^^^^^
+
+Add or remove an address as operator of the address sending this message.
+
+Parameter
+~~~~~~~~~
+
+The parameter contains whether to add or remove an operator and the address to add/remove as operator.
+It does not contain the address which are adding/removing the operator as this will be the sender of the message invoking this function.
```

Comment:
```suggestion
It does not contain the address which is adding/removing the operator as this will be the sender of the message invoking this function.
```